devel/README.md

```diff
@@ -0,0 +1,54 @@
+# Developing on Istio
+
+See also the [broker](https://github.com/istio/broker/blob/master/doc/dev/development.md) and [mixer](https://github.com/istio/mixer/blob/master/doc/dev/development.md)
+development setup and guidelines
+
+## Collection of scripts and notes for developing on Istio
+
+For local development (building from source and running the major components) on Ubuntu/raw VM:
+
+Assuming you did (once):
+1. [Install bazel](https://bazel.build/versions/master/docs/install-ubuntu.html), note that as of this writing Bazel needs the `openjdk-8-jdk` VM (you might need to uninstall or get out of the way the `ibm-java80-jdk` that comes by default with GCE for instance)
+2. Install required packages: `sudo apt-get install make openjdk-8-jdk libtool m4 autoconf uuid-dev`
+3. Get the source trees
+   ```bash
+   mkdir github
+   cd github/
+   git clone https://github.com/istio/proxy.git
+   git clone https://github.com/istio/mixer.git
+   git clone https://github.com/istio/istio.git
+   # if you want to do load tests:
+   git clone https://github.com/wg/wrk.git
+   ```
+4. You can then use 
+   - [update_all](update_all) : script to build from source 
+   - [setup_run](setup_run)  : run locally
+   - Also found in this directory: [rules.yml](rules.yml) : the version of  mixer/testdata/configroot/scopes/global/subjects/global/rules.yml that works locally and [quota.yml](quota.yml) a very simple 1 qps quota example used below.
+5. And run things like
+   ```bash
+   # Test the echo server:
+   curl -v http://localhost:8080/
+   # Test through the proxy:
+   curl -v http://localhost:9090/echo
+   # Add a rule locally (simply drop the file or exercise the API:)
+   curl -v  http://localhost:9094/api/v1/scopes/global/subjects/foo.svc.cluster.local/rules --data-binary @quota.yaml -X PUT -H "Content-Type: application/yaml"
+   # Test under some load:
+   wrk http://localhost:9090/echo
+   
+   ```
+   Note that this is done for you by [steup_run](setup_run) but to use the correct go environment:
+   ```bash
+   cd mixer/
+   source bin/use_bazel_go.sh 
+   ```
+
+
+## MacOs tips
+ 
+Get GitHub desktop https://desktop.github.com/
+ 
+If you want to make changes to the [website](https://github.com/istio/istio.github.io), and want to run jekyll locally and natively, without docker):
```

Comment:
well it's much lighter running natively and only 1 paragraph, I had that in my notes and think it may be useful to someone else - I'd like eventually to run everything natively on the mac btw (low pri but still)